src/main/java/uk/gov/hmcts/dts/fact/model/Court.java

```diff
@@ -45,25 +51,77 @@
 
     public Court(uk.gov.hmcts.dts.fact.entity.Court courtEntity) {
         this.name = courtEntity.getName();
         this.slug = courtEntity.getSlug();
-        this.info = courtEntity.getInfo();
+        this.info = stripHtmlFromString(courtEntity.getInfo());
         this.open = courtEntity.getDisplayed();
         this.directions = courtEntity.getDirections();
+        this.imageFile = courtEntity.getImageFile();
         this.lat = courtEntity.getLat();
         this.lon = courtEntity.getLon();
+        this.alert = courtEntity.getAlert();
         this.crownLocationCode = courtEntity.getNumber();
         this.countyLocationCode = courtEntity.getCciCode();
         this.magistratesLocationCode = courtEntity.getMagistrateCode();
-        this.areasOfLaw = courtEntity.getAreasOfLaw().stream().map(AreaOfLaw::getName).collect(toList());
+        this.areasOfLaw = courtEntity.getAreasOfLaw().stream().map(AreaOfLaw::new).collect(toList());
         this.courtTypes = courtEntity.getCourtTypes().stream().map(CourtType::getName).collect(toList());
         this.emails = courtEntity.getEmails().stream().map(CourtEmail::new).collect(toList());
         this.contacts = courtEntity.getContacts().stream().map(Contact::new).collect(toList());
         this.openingTimes = courtEntity.getOpeningTimes().stream().map(OpeningTime::new).collect(toList());
-        this.facilities = courtEntity.getFacilities().stream().map(Facility::new).collect(toList());
-        this.addresses = courtEntity.getAddresses().stream().map(CourtAddress::new).collect(toList());
+        this.facilities = this.stripHtmlFromFacilities(
+            courtEntity.getFacilities().stream().map(Facility::new).collect(toList()));
+        this.addresses = this.refactorAddressType(
+            courtEntity.getAddresses().stream().map(CourtAddress::new).collect(toList()));
         this.gbs = courtEntity.getGbs();
+        this.dxNumber = this.getDxNumber(this.contacts);
+        this.serviceArea = courtEntity.getServiceArea() == null ? null : courtEntity.getServiceArea().getService();
+        this.inPerson = courtEntity.getInPerson() == null ? null : courtEntity.getInPerson().getIsInPerson();
+    }
+
+    private List<String> getDxNumber(List<Contact> contacts) {
+        List<Contact> dx = contacts
+            .stream()
+            .filter(c -> "DX".equals(c.getName()))
+            .collect(toList());
+        this.contacts.removeAll(dx);
+        return dx
+            .stream()
+            .map(Contact::getNumber)
+            .collect(toList());
+    }
+
+    private List<Facility> stripHtmlFromFacilities(List<Facility> facilities) {
+        for (Facility facility : facilities) {
+            facility.setDescription(stripHtmlFromString(facility.getDescription()));
+        }
+        return facilities;
+    }
+
+    private String stripHtmlFromString(String text) {
+        if (text != null) {
+            return text.replaceAll("\\<.*?>|&nbsp;|amp;", "");
```

Comment:
This looks like it might get re-used to, perhaps split it out somewhere as a static method

Reply:
Same here